src/utils/general.jl

```diff
@@ -117,8 +117,11 @@ An accessor primarily written for accessing fields of LSP objects whose fields
 may be unset (set to `nothing`) depending on client/server capabilities.
 Traverses the field chain `paths...` of `obj`, and returns `nothing` if any
 `nothing` field is encountered along the way.
+
+Note: `@noinline` is used to maximize type stability. This is a temporary workaround and
+may become unnecessary with future compiler improvements.
```

Comment:
It might be best to add test case like
```julia
julia> @test Base.infer_return_type((typeof(manager),)) do manager
           JETLS.get_config(manager, :full_analysis, :debounce)
       end == Union{Nothing,Float64}
```